src/nvim/undo.c

```diff
@@ -1487,7 +1498,83 @@ void u_read_undo(char_u *name, char_u *hash, char_u *orig_name)
   return;
 }
 
+/// Writes a sequence of bytes to the undo file.
+///
+/// @returns false in case of an error.
+static bool undo_write(bufinfo_T *bi, uint8_t *ptr, size_t len)
+{
+  return fwrite(ptr, len, 1, bi->bi_fp) == 1;
+}
+
+/// Writes a number, MSB first, in "len" bytes.
+///
+/// Must match with undo_read_?c() functions.
+///
+/// @returns false in case of an error.
+static bool undo_write_bytes(bufinfo_T *bi, uintmax_t nr, size_t len)
+{
+  assert(len > 0);
+  uint8_t buf[8];
+  for (size_t i = len - 1, bufi = 0; i < len; i--, bufi++) {
+    buf[bufi] = (uint8_t)(nr >> (i * 8));
+  }
+  return undo_write(bi, buf, len);
+}
 
+/*
+ * Write the pointer to an undo header.  Instead of writing the pointer itself
+ * we use the sequence number of the header.  This is converted back to
+ * pointers when reading. */
```

Comment:
Since this appears as changed in the diff might as well convert the comment to `///` Doxygen comment.